x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/EsqlParser.java

```diff
@@ -52,20 +54,27 @@ public void setEsqlConfig(EsqlConfig config) {
         this.config = config;
     }
 
+    // testing utility
     public LogicalPlan createStatement(String query) {
         return createStatement(query, new QueryParams());
     }
 
+    // testing utility
     public LogicalPlan createStatement(String query, QueryParams params) {
+        return createStatement(query, params, new PlanningMetrics(new EsqlFunctionRegistry()));
```

Comment:
Where is this used - I'd rather force consumer to change the constructor instead of hiding this object creation underneath.

Reply:
It's only used in tests (I'd added a comment). 
This could be moved out, but subsequently, as there are plenty of tests not caring about params or metrics.